project-addons/sale_custom/models/sale.py

```diff
@@ -34,3 +34,29 @@ def product_id_change(self, pricelist, product, qty=0,
             date_order=date_order, packaging=packaging,
             fiscal_position=fiscal_position, flag=flag)
 
+
+class SaleOrder(models.Model):
+
+    _inherit = "sale.order"
+
+    @api.multi
+    def open_historical_orders(self):
+        self.ensure_one()
+        partner_id = self.partner_id.commercial_partner_id.id
+        order_view_id = self.env.ref('sale.act_res_partner_2_sale_order').id
+        last_order = self.env['sale.order'].search([('id', '!=', self.id),
+                                                    ('partner_id', '=', partner_id),
+                                                    ('state', 'not in', ['cancel', 'draft', 'sent'])],
+                                                   limit=1, order='date_order DESC').id
```

Comment:
¿Sólo queréis abrir una?

Reply:
Sí, se tiene que abrir el último pedido del cliente, pero a su vez tener accesible el resto de pedidos por si necesitan consultar algún otro.

Reply:
Ok, y ¿cómo tienes accesible el resto de pedidos?
Tened en cuenta que si lo hacéis en el propio Odoo, con una acción de venta, luego con las migas de pan volvéis al pedido en el que estabas sin ningún problema, es decir se podrían crear dos botones "smart button" que le llaman, como los que hay en la vista de empresas o productos arriba a la derecha uno para que muestre el último pedido (método normal en sale.order que retorna una acción de venta) y otro para ver todos los pedidos y sería lo mismo  (método normal en sale.order que retorna una acción de venta) ven lo que necesiten, clikan en las migas de pan del pedido en el que estaban y ya estaría. No lo tienen en una pestaña nueva, pero es perfectamente navegable y con los mismos clicks, lo digo porque es una funcionalidad más nativa, que seguro que sigue funcionando en próximas versiones y esto quizás no, pero como veáis, quitando la corrección que pongo en el otro comentarios técnicamente estaría bien.

Reply:
El problema es que necesitan tenerlo en otra pestaña para poder seguir editando el pedido que estén registrando. De hecho, nuestra primera opción era mostrar la lista de pedidos anteriores en un pop-up, pero al inhabilitar la edición del pedido que están creando, no nos servía.

Reply:
Ok, mergeo así